src/main/java/entities/Comment.java

```diff
@@ -5,32 +5,62 @@
 import java.util.Date;
 import java.util.List;
 
-import java.security.PublicKey;
-
-
+/**
+ * Comment extends abstract class Postable and inherits all of its
+ * instance attributes (user, body, replies) and addComment method.
+ *
+ */
 public class Comment extends Postable{
-    // Class Comment extends abstract class Postable and inherits all of its instance
-    // attributes (user, body, replies) and addComment method
-    // (which has already been implemented)
-    public static Date creationDate;
+    // number of Comments instantiated
+    public static int numComments = 0;
+    // id is the unique identifier of Comment, equal to number of Comments at time of instantiation
+    private final int id;
 
-    public Comment(User commenter, String body){
+    /**
+     * Initializes an instance of Comment.
+     *
+     * @param commenter The user that posted this comment.
+     * @param body The text content of the comment.
+     * @param numCommentsCreated The number of comments created in the program so far.
+     */
+    public Comment(User commenter, String body, int numCommentsCreated){
         super.user = commenter;
         super.body = body;
         super.replies = new ArrayList<>();
-        creationDate = new Date();
-
+        super.creationDate = new Date();
+        Comment.numComments = numCommentsCreated;
+        Comment.numComments ++;
+        this.id = Comment.numComments;
     }
+
+    /**
+     * Returns the body (or text content) of this instance of Comment.
+     */
     public String getBody(){
```

Comment:
This method seems like something that the abstract class should already have implemented. We could change it to just be a method in super and then Comment and Post would already have these implemented

Reply:
moved!